### PR TITLE
Mushr feedback control

### PR DESCRIPTION
**Pull request by aravindsiv**
Adding a single feedback controller class that can use a controller implemented inside ML4KP.

### CODE REVIEW

File: control/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+)
+
+###########
+## Build ##
+###########
+
+## Specify additional locations of header files
+## Your package locations should be listed before other locations
+include_directories(
+  include
```
Comment (garygra) [inline control/CMakeLists.txt]: No include directory, remove this line unless there is a missing file.

File: control/src/nodelets/mushr_feedback_controller.cpp [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    else if (controller_params["name"].as<std::string>() == "mushr_stanley")
+    {
+      controller = new prx::mushr_stanley_t(plant, controller_params);
+    }
+    else
+    {
+      ROS_ERROR("Unknown controller type: %s", controller_params["name"].as<std::string>().c_str());
+      return;
+    }
+    controller->set_goal(ml4kp_params["goal_state"].as<std::vector<double>>());
```
Comment (garygra) [inline control/src/nodelets/mushr_feedback_controller.cpp:67]: There is no templated set_goal in the main branch of ML4KP, is there a PR missing?

File: control/src/nodelets/mushr_feedback_controller.cpp [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  {
+    prx_models::copy(current_state, msg);
+    _pose_received = true;
+  }
+
+  void reset(const std_msgs::Empty& msg)
+  {
+    _trajectory_received = false;
+    _pose_received = false;
+    controller->reset();
```
Comment (garygra) [inline control/src/nodelets/mushr_feedback_controller.cpp:111]: Same as above, no reset on `ML4KP-devel`'s main

File: control/src/nodelets/mushr_feedback_controller.cpp [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  }
+
+  void control(const ros::TimerEvent& event)
+  {
+    controller->get_control(current_state, current_control);
```
Comment (garygra) [inline control/src/nodelets/mushr_feedback_controller.cpp:116]: `get_control` is only defined for `mushr_pure_persuit`, not for the feedback controller.

**Review by garygra — COMMENTED** [review]
Need some changes because is not compiling as is.

**Comment by aravindsiv** [comment]
Let me send a PR for `ML4KP-devel` to compile this PR by tonight.

**Comment by aravindsiv** [comment]
[This PR](https://github.com/PRX-Kinodynamic/ML4KP-devel/pull/43) on `ML4KP-devel` should help with compiling this one.

File: control/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+## catkin specific configuration ##
+###################################
+## The catkin_package macro generates cmake config files for your package
+## Declare things to be passed to dependent projects
+## INCLUDE_DIRS: uncomment this if your package contains header files
+## LIBRARIES: libraries you create in this project that dependent projects also need
+## CATKIN_DEPENDS: catkin_packages dependent projects also need
+## DEPENDS: system dependencies of this project that dependent projects also need
+catkin_package(
+ INCLUDE_DIRS include
```
Comment (garygra) [inline control/CMakeLists.txt]: Comment this out, no `include` dir on control/

**Review by garygra — CHANGES_REQUESTED** [review]
Minor problem on `control/CMakeLists.txt`. Otherwise is compiling.

**Review by garygra — COMMENTED** [review]
Removing INCLUDE_DIRS include

**Review by garygra — APPROVED** [review]

**Pull request merged**